### PR TITLE
refactor: Reuse lens wasm modules

### LINKED ISSUE #1640: Reuse lens wasm modules
Part of https://github.com/sourcenetwork/defradb/issues/1002

Using new Lens code: https://github.com/lens-vm/lens/compare/49-runtime-rework?expand=1

### PR DESCRIPTION
**Pull request by AndrewSisley**
## Relevant issue(s)

Resolves #1640

## Description

Reuses lens wasm modules to spawn new module instances.  Is an awful lot cheaper - I span up 1000 duplicates in defra and the cost was too small for me to bother measuring it properly, might have been 1-3GB (for the 1000 instances) (compared to 20GB or so previously).

Draft PR as it currently is pointed at a local path, new Lens code can be found here: https://github.com/lens-vm/lens/pull/50.
- [x] Update Lens dependency

### CODE REVIEW

File: lens/registry.go [hunk 1]
```diff
@@ -246,10 +260,11 @@ type lensPool struct {
 	pipes chan *lensPipe
 }
 
-func newLocker(lensPoolSize int, cfg client.LensConfig) *lensPool {
+func (r *lensRegistry) newLocker(lensPoolSize int, cfg client.LensConfig) *lensPool {
```
[the remark this answers is not in the document]
Comment (AndrewSisley) [reply lens/registry.go:263]: Cheers :) Must have been missed when I renamed all of these things in the original PR

- [ ] Rename newLocker

**Review by jsimnz — APPROVED** [review]
Lgtm

**Review by fredcarle — APPROVED** [review]
LGTM

**Review by shahzadlone — APPROVED** [review]
LGTM

**Pull request merged**